### PR TITLE
Support create/delete veth interface

### PR DESCRIPTION
**Pull request by cathay4t**
Example yaml for `npc set`:

```yml
---
ifaces:
  - name: veth1
    type: veth
    veth:
      peer: veth1.ep
    ipv4:
      addresses:
        - address: "192.0.2.1"
          prefix_len: 24
    ipv6:
      addresses:
        - address: "2001:db8:a::9"
          prefix_len: 64
```

To delete a interface, you may:

```yml
---
ifaces:
  - name: veth1
    state: absent
```

The creating of 1000 veth interface(including another 1000 veth peer)
takes 5.120 seconds.

Integration tests for creating and delete veth have been included.

### CODE REVIEW

**Comment by cathay4t** [comment]
@ffmancera Patch ready for review.

**Comment by ffmancera** [comment]
Big patch, taking a lot. I will let you know tomorrow my thoughts. Thanks!

**Comment by cathay4t** [comment]
The performance is bad(8m33.713s) when trying to create 1000 veth pairs. Will improve it.

File: src/lib/ip.rs [hunk 1]
```diff
@@ -61,6 +62,37 @@ pub struct IpConf {
...
+                addrs.push(IpAddrConf {
+                    address: addr_info.address.clone(),
+                    prefix_len: addr_info.prefix_len,
+                });
+            }
+        }
+        Self { addresses: addrs }
+    }
+}
+
```
Comment (qinqon) [inline src/lib/ip.rs:94]: Can we make a ipv4/ipv6 generic function for this ?

File: src/lib/ip.rs [hunk 2]
```diff
@@ -81,141 +113,254 @@ impl IpConf {
...
+                    prefix_len: des_addr.prefix_len,
+                });
+            }
+            for cur_addr in &cur_ip_conf.addresses {
+                cur_ip_addr_confs.insert(IpAddrConf {
+                    address: cur_addr.address.clone(),
+                    prefix_len: cur_addr.prefix_len,
+                });
+            }
+            let has_ipv6_link_local_in_desire = if ip_family == IpFamily::Ipv4 {
```
Comment (qinqon) [inline src/lib/ip.rs:301]: This does not have to be `ip_family == IpFamily::Ipv4`

**Review by qinqon — CHANGES_REQUESTED** [review]
First pass, a pair of small things/questions

File: src/lib/ip.rs (hunk 1 above)
Comment (cathay4t) [reply src/lib/ip.rs:94]: They are taking different type of input. Maybe [`Box<dyn Trait>`][1] might help, but I don't like to include complexity which is unfriendly to me and other new rust learner. If you have suggestions, please kindly let me know.


[1]: https://doc.rust-lang.org/edition-guide/rust-2018/trait-system/dyn-trait-for-trait-objects.html

File: src/lib/ip.rs (hunk 2 above)
Comment (cathay4t) [reply src/lib/ip.rs:301]: It saves some cpu time on iteration of `is_ipv6_unicast_link_local_full` over ipv4 addresses.

**Comment by cathay4t** [comment]
@ffmancera Patch ready for review.

**Comment by cathay4t** [comment]
Let's merge this and get it going. Other PR is waiting this.

**Pull request merged**